src/MySqlConnector/Logging/ConsoleLoggerProvider.cs

```diff
@@ -0,0 +1,94 @@
+using System;
+using System.Text;
+
+namespace MySqlConnector.Logging
+{
+	public class ConsoleLoggerProvider : IMySqlConnectorLoggerProvider
+	{
+		public ConsoleLoggerProvider(MySqlConnectorLogLevel minimumLevel = MySqlConnectorLogLevel.Info, bool isColored = true)
+		{
+			if (minimumLevel < MySqlConnectorLogLevel.Trace || minimumLevel > MySqlConnectorLogLevel.Fatal)
+				throw new ArgumentOutOfRangeException(nameof(minimumLevel), "minimumLevel must be between Trace and Fatal");
+
+			m_minimumLevel = minimumLevel;
+			m_isColored = isColored;
+		}
+
+		public IMySqlConnectorLogger CreateLogger(string name) => new ConsoleLogger(this, name);
+
+		private sealed class ConsoleLogger : IMySqlConnectorLogger
+		{
+			public ConsoleLogger(ConsoleLoggerProvider provider, string name)
+			{
+				m_provider = provider;
+				m_name = name;
+			}
+
+			public bool IsEnabled(MySqlConnectorLogLevel level) => level >= m_provider.m_minimumLevel && level <= MySqlConnectorLogLevel.Fatal;
+
+			public void Log(MySqlConnectorLogLevel level, string message, object[] args = null, Exception exception = null)
+			{
+				if (!IsEnabled(level))
+					return;
+
+				var sb = new StringBuilder();
+				sb.Append(s_levels[(int) level]);
+				sb.Append('\t');
+				sb.Append(m_name);
+				sb.Append('\t');
+
+				if (args == null)
+					sb.Append(message);
+				else
+					sb.AppendFormat(message, args);
```

Comment:
I'd use `CultureInfo.InvariantCulture` here.

Reply:
If only to demonstrate how to correctly write a logging implementation, sure.